crates/xilem_masonry/src/view/button.rs

```diff
@@ -53,4 +44,22 @@ where
             ChangeFlags::UNCHANGED
         }
     }
+
+    fn message(
+        &self,
+        _view_state: &mut Self::ViewState,
+        id_path: &[ViewId],
+        message: Box<dyn std::any::Any>,
+        app_state: &mut State,
+    ) -> crate::MessageResult<Action> {
+        debug_assert!(
+            id_path.is_empty(),
+            "id path should be empty in Button::message"
+        );
+        if let Some(masonry::Action::ButtonPressed) = message.downcast_ref() {
+            return MessageResult::Action((self.callback)(app_state));
+        }
+        tracing::error!("Wrong message type in Button::message");
```

Comment:
Ideally, we'd be using a `Box<dyn Any + Debug>` or something similar, to make this more debuggable. But that's also out-of-scope for this PR